### ISSUE TITLE
`no-trailing-spaces`: should report trailing spaces at the end of lines immediately followed by another line?

### ISSUE THREAD

**Issue by andreww2012** [ota-meshi/eslint-plugin-markdown-preferences#234]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.39.2

### What version of `eslint-plugin-markdown-preferences` are you using?

0.40.1

### What did you do?

I'm not sure if this is intentional, but trailing spaces at the end of lines after which another non-empty line goes are not reported:

```md
foo 
bar
```

### What did you expect to happen?

I think it is safe to remove the space after `foo` in the example above. A space is automatically inserted in the output.

### What actually happened?

It is not reported.

### Link to **GitHub Repo** with Minimal Reproducible Example

https://eslint-online-playground.netlify.app/#eNp9Ustu6yAQ/ZUpq0SKyT739iG1XbeLSl2ELIgZO07xYAFuk0b59w7E7ktRJYTNcM6ZmTMcRPDlHHe67SzK1oiF+H9RFPC82UPcNAGi141tqIbQ6RKBI+QieOycj2iuoSiuFCmqnANFa+3TISs8JTavEQprLHUfMPMrZ617S7Jxg6A5tEEPnAdHwbV+Z0ExExg4HGXpqGpq2W4Dl9i0SRMOYLBizm2+gyNU3rWgBsr8RFHin6L5HAZOq/2LcW80Ym8G8BjP8F/YR48VeqQSw88URWf7uqFiBBbdFzILcQmB3eot3lP0DfMv4WG9xTJKPAUmZ5LIRAhTWTU2op8oApgsZ1lmNYXLK7hIv7LFqKVBzllqNnimaMopcZdrZ2d0b/P306HJMkkd0gaAu4hkwgKW5yo4eRdkiJqM9mbF6ollNdW9rnHBHpzrmp9SkkVThD1FvVMiE4+8r1J1PFB+Ry+sILfBEc8yl6NE3HeoRJZ1hrsbiEoYfL3DLmlSyX4lzNDBueElActuhDgIJNgJ9efl35P8TsztKOIljh9qBCSN
